services/QuillLMS/engines/evidence/app/models/evidence/research/gen_ai/experiment.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: evidence_research_gen_ai_experiments
+#
+#  id                :bigint           not null, primary key
+#  experiment_errors :text             default([]), not null, is an Array
+#  results           :jsonb
+#  status            :string           default("pending"), not null
+#  created_at        :datetime         not null
+#  updated_at        :datetime         not null
+#  llm_config_id     :integer          not null
+#  llm_prompt_id     :integer          not null
+#  passage_prompt_id :integer          not null
+#
+module Evidence
+  module Research
+    module GenAI
+      class Experiment < ApplicationRecord
+        STATUSES = [
+          PENDING = 'pending',
+          RUNNING = 'running',
+          COMPLETED = 'completed',
+          FAILED = 'failed'
+        ].freeze
+
+        belongs_to :llm_config, class_name: 'Evidence::Research::GenAI::LLMConfig'
+        belongs_to :llm_prompt, class_name: 'Evidence::Research::GenAI::LLMPrompt'
+        belongs_to :passage_prompt, class_name: 'Evidence::Research::GenAI::PassagePrompt'
+
+        has_many :passage_prompt_responses,
+          class_name: 'Evidence::Research::GenAI::PassagePromptResponse',
+          through: :passage_prompt
+
+        validates :llm_config_id, :llm_prompt_id, :passage_prompt_id, presence: true
+        validates :status, presence: true, inclusion: { in: STATUSES }
+
+        delegate :conjunction, :name, to: :passage_prompt
+        delegate :llm_client, to: :llm_config
+        delegate :vendor, :version, to: :llm_config
+
+        attr_accessor :llm_prompt_template_id
+
+        def completed! = update!(status: COMPLETED)
+        def running! = update!(status: RUNNING)
+        def failed! = update!(status: FAILED)
+
+        def run
+          running!
```

Comment:
Should a non-`pending` state block this code from running?

Reply:
Good idea.